commit 227e2a05ac94d11fb4002b28d33e8619414215d4
Author: Annaliese McDermond <nh6z@nh6z.net>
Date:   Thu Mar 21 17:58:52 2019 -0700

    ASoC: tlv320aic32x4: Restructure set_dai_sysclk
    
    commit aa6a60f7be925210d5156f0e8025f3afe1f4f54d upstream.
    
    The sysclk is now managed by the CCF.  Change this function
    to merely find the system clock and set it using
    clk_set_rate.
    
    Signed-off-by: Annaliese McDermond <nh6z@nh6z.net>
    Signed-off-by: Mark Brown <broonie@kernel.org>

diff --git a/sound/soc/codecs/tlv320aic32x4.c b/sound/soc/codecs/tlv320aic32x4.c
index 378696e80627..41778ad7c668 100644
--- a/sound/soc/codecs/tlv320aic32x4.c
+++ b/sound/soc/codecs/tlv320aic32x4.c
@@ -49,7 +49,6 @@
 
 struct aic32x4_priv {
 	struct regmap *regmap;
-	u32 sysclk;
 	u32 power_cfg;
 	u32 micpga_routing;
 	bool swapdacs;
@@ -567,17 +566,13 @@ static int aic32x4_set_dai_sysclk(struct snd_soc_dai *codec_dai,
 				  int clk_id, unsigned int freq, int dir)
 {
 	struct snd_soc_component *component = codec_dai->component;
-	struct aic32x4_priv *aic32x4 = snd_soc_component_get_drvdata(component);
+	struct clk *mclk;
+	struct clk *pll;
 
-	switch (freq) {
-	case 12000000:
-	case 24000000:
-	case 25000000:
-		aic32x4->sysclk = freq;
-		return 0;
-	}
-	printk(KERN_ERR "aic32x4: invalid frequency to set DAI system clock\n");
-	return -EINVAL;
+	pll = devm_clk_get(component->dev, "pll");
+	mclk = clk_get_parent(pll);
+
+	return clk_set_rate(mclk, freq);
 }
 
 static int aic32x4_set_dai_fmt(struct snd_soc_dai *codec_dai, unsigned int fmt)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1411_227e2a05ac94d11fb4002b28d33e8619414215d4.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1411_227e2a05ac94d11fb4002b28d33e8619414215d4.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
